server/src/main/java/com/broadcom/lsp/cobol/core/engine/CobolLanguageEngine.java

```diff
@@ -157,7 +160,42 @@ CobolParser getCobolParser(CommonTokenStream tokens) {
 
   @NonNull
   private Function<SyntaxError, SyntaxError> convertError(@NonNull Map<Token, Locality> mapping) {
-    return err -> err.toBuilder().locality(mapping.get(err.getOffendedToken())).build();
+    return err ->
+        err.toBuilder()
+            .locality(
+                mapping.getOrDefault(
+                    err.getOffendedToken(), getNearestLocality(err.getOffendedToken(), mapping)))
```

Comment:
Another controversial change. 
Reason: In case we want to highlight an error from a lexer action based on tokens, which is a subset of a single token.
For e.g. suppose we have a single token, SINGLETOKEN, and based on lexer action, it is decided that there is an error on the "LE" part of that token. In this case, we want to highlight the entire token with an error message generated by the lexer action.